sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -2166,6 +2166,22 @@ abstract class DDLSuite extends QueryTest with DDLSuiteBase {
       assert(spark.sessionState.catalog.isRegisteredFunction(rand))
     }
   }
+
+  test("NOT_A_PARTITIONED_TABLE: ALTER TABLE is not allowed for non-partitioned table") {
```

Comment:
Precisely speaking,  `ALTER TABLE .. RECOVER PARTITIONS` not just `ALTER TABLE`

Reply:
Just moved the test into dedicated test suites, and update the test name.
Thanks!